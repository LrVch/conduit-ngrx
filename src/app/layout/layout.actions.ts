import { Action } from '@ngrx/store';

export enum LayoutActionTypes {
  ShowMainLoader = '[Layout] Show Main Loader',
  HideMainLoader = '[Layout] Hide Main Loader',
  ToggleSideNav = '[Layout] Toggle Side Nav',
}

export class ShowMainLoader implements Action {
  readonly type = LayoutActionTypes.ShowMainLoader;
}

export class HideMainLoader implements Action {
  readonly type = LayoutActionTypes.HideMainLoader;
}

export class ToggleSideNav implements Action {
  readonly type = LayoutActionTypes.ToggleSideNav;

  constructor(readonly payload?: { close?: boolean }) { }
}


export type LayoutActions =
  ShowMainLoader
  | HideMainLoader
  | ToggleSideNav;
