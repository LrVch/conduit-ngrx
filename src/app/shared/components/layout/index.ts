export * from './footer/footer.component';
export * from './header/header.component';
export * from './page-not-found/page-not-found.component';
export * from './aside/aside.component';
