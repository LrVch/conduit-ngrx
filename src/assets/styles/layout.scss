* {
    box-sizing: border-box;
    &:focus,
    &:active {
        outline: none !important;
    }
}

html,
body {
    height: 100%;
}

body {
    min-width: 320px;
}

.sticky-header {
    header {
        position: fixed !important;
    }
    .layout {
        padding-top: 64px;
        @media (max-width: 767px) {
            padding-top: 96px;
        }
    }
    .header {
        .app-preloader {
            position: absolute !important;
            top: 100% !important;
        }
    }
}

.layout {
    height: 100%;
}

.main {
    display: flex;
    flex-direction: column;
    height: 100%;
    overflow-y: scroll;
}

.main-content {
    position: relative;
    flex-grow: 1;
}

.content {
    flex-grow: 1;
}

.banner {
    padding-top: 40px;
    padding-bottom: 40px;
}

a.logo {
    @media (max-width: 599px) {
        font-size: 15px;
    }
}

.content {
    margin-top: 25px;
    padding-bottom: 10px;
}

.pagination {
    display: flex;
    justify-content: flex-end;
    margin-top: 20px;
}

.tags-wtapper {
    position: sticky;
    top: 34px;
    @media (max-width: 991px) {
        margin-bottom: 30px;
        position: static;
    }
}

.page {
    padding-top: 100px;
    @media (max-width: 1023px) {
        padding-top: 50px;
    }
    @media (max-width: 599px) {
        padding-top: 30px;
    }
}

.input-full-width {
    width: 100%;
}

.loader {
    display: block;
    width: 100%;
}

.img-loading {
    opacity: 0;
    transition: opacity .3s;
}

.img-loaded-ok {
    transition: opacity 0;
    opacity: 1;
}

.img-loaded {
    opacity: 1;
}

.icon-form-field {
    display: flex;
    align-items: flex-start;
    margin-bottom: 10px;
    &:last-child {
        margin-bottom: 0;
    }
    mat-icon {
        margin-right: 5px;
        margin-bottom: 2px;
    }
    mat-placeholder {
        flex-grow: 1;
        padding-right: 5px;
    }
}

.icon-form-field_select {
    align-items: center;
    mat-icon {
        margin-bottom: 7px;
    }
}

.color-view {
    display: flex;
    align-items: center;
}

.color-view__label {
    flex-grow: 1;
}

.color-view__color {
    width: 24px;
    height: 24px;
    border-radius: 4px;
}