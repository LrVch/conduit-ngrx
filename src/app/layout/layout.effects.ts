import { Injectable, NgZone } from '@angular/core';
import { Actions, Effect } from '@ngrx/effects';
import { Router, NavigationEnd } from '@angular/router';
import { Store } from '@ngrx/store';
import { AppState } from '@app/reducers';
import { filter, tap } from 'rxjs/operators';
import {} from './layout.selectors';
import { ScrollService } from '@app/core';

@Injectable()
export class LayoutEffects {
  constructor(
    private actions$: Actions,
    private router: Router,
    private ngZone: NgZone,
    private scrollService: ScrollService
  ) {}

  @Effect({ dispatch: false })
  scrollPageToTop$ = this.router.events
    .pipe(filter(event => event instanceof NavigationEnd))
    .pipe(
      tap(() => {
        this.ngZone.runOutsideAngular(() => {
          this.scrollService.scrollMainContainerTo();
        });
      })
    );
}
