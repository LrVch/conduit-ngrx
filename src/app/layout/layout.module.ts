import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { StoreModule } from '@ngrx/store';
import * as fromLayout from './layout.reducer';
import { EffectsModule } from '@ngrx/effects';
import { LayoutEffects } from './layout.effects';

@NgModule({
  imports: [
    CommonModule,
    StoreModule.forFeature('layout', fromLayout.layoutReducer),
    EffectsModule.forFeature([LayoutEffects])
  ],
  declarations: []
})
export class LayoutModule { }
