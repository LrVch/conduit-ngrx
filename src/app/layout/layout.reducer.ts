import { Action } from '@ngrx/store';
import { LayoutActionTypes } from './layout.actions';


export interface LayoutState {
  showMainLoader: boolean;
}

export const initialState: LayoutState = {
  showMainLoader: false
};

export function layoutReducer(state = initialState, action: Action): LayoutState {
  switch (action.type) {
    case LayoutActionTypes.ShowMainLoader:
      return {
        ...state,
        showMainLoader: true
      };
    case LayoutActionTypes.HideMainLoader:
      return {
        ...state,
        showMainLoader: false
      };
    default:
      return state;
  }
}
