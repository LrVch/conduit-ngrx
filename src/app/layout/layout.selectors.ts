import { createSelector, createFeatureSelector } from '@ngrx/store';
import { AppState } from '@app/reducers';
import { LayoutState } from './layout.reducer';

export const selectLayoutState = createFeatureSelector<AppState, LayoutState>('layout');

export const selectShowMainLoader = createSelector(
  selectLayoutState,
  (layoutState: LayoutState) => layoutState.showMainLoader
);
