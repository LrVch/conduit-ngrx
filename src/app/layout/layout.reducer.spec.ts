import * as fromLayout from './layout.reducer';
import { ShowMainLoader, HideMainLoader } from './layout.actions';

describe('Auth Reducer', () => {
  const initialAuthState: fromLayout.LayoutState = {
    showMainLoader: false
  };

  describe('State changes', () => {
    it('should have an initial state', () => {
      const state = fromLayout.layoutReducer(initialAuthState, { type: '@@init' } as any);

      expect(state).toBe(initialAuthState);
    });

    it('should show main loader', () => {
      const init = { type: '@@init' } as any;
      const loginSuccess = new ShowMainLoader();

      const state = [init, loginSuccess].reduce(fromLayout.layoutReducer, initialAuthState);

      (<any>expect(state)).toMatchSnapshot();
    });

    it('should hide main loader', () => {
      const init = { type: '@@init' } as any;
      const loginSuccess = new HideMainLoader();

      const state = [init, loginSuccess].reduce(fromLayout.layoutReducer, initialAuthState);

      (<any>expect(state)).toMatchSnapshot();
    });
  });
});
