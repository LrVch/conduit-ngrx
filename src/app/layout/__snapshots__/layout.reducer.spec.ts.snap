// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Reducer State changes should hide main loader 1`] = `
Object {
  "showMainLoader": false,
}
`;

exports[`Auth Reducer State changes should show main loader 1`] = `
Object {
  "showMainLoader": true,
}
`;
